test/container/php/test_fpm_entrypoint.py

```diff
@@ -11,3 +11,67 @@ def test_php_fpm_exec_has_dumb_init(host):
     assert "--rewrite 1:17" in php_fpm_exec.stdout
 
     assert "shush" not in php_fpm_exec.stdout
+
+@pytest.mark.php_fpm_exec
+def test_php_fpm_templates_config_file(host):
+    config_file = host.file("/usr/local/etc/php-fpm.d/zz-docker.conf")
+
+    assert config_file.exists is True
+    assert config_file.is_file is True
+
+    # Test default configuration for FPM PM
+    config = config_file.content
+
+    assert "pm = dynamic" in config
+    assert "pm.max_children = 5" in config
+    assert "pm.start_servers = 2" in config
+    assert "pm.min_spare_servers = 1" in config
+    assert "pm.max_spare_servers = 3" in config
+    assert "pm.process_idle_timeout = 10" in config
+    assert "pm.max_requests = 0" in config
+
+    # Test effective configuration
+    config_effective = host.check_output("php-fpm -tt")
+
+    assert "pm = dynamic" in config_effective
+    assert "pm.max_children = 5" in config_effective
+    assert "pm.start_servers = 2" in config_effective
+    assert "pm.min_spare_servers = 1" in config_effective
+    assert "pm.max_spare_servers = 3" in config_effective
+    assert "pm.process_idle_timeout = 10" in config_effective
+    assert "pm.max_requests = 0" in config_effective
+
+@pytest.mark.php_fpm_exec
+def test_php_fpm_templates_config_file_all_vars(host):
+    config_template = host.file("/usr/local/etc/php-fpm.d/zz-docker.conf.template")
+
+    assert config_file.exists is True
+    assert config_file.is_file is True
+
+    replace_template = host.run('''
+    ENV PHP_FPM_PM=static \
+    ENV PHP_FPM_PM_MAX_CHILDREN=70 \
+    ENV PHP_FPM_PM_START_SERVERS=10 \
+    ENV PHP_FPM_PM_MIN_SPARE_SERVERS=20 \
+    ENV PHP_FPM_PM_MAX_SPARE_SERVERS=40 \ 
+    ENV PHP_FPM_PM_PROCESS_IDLE_TIMEOUT=35 \
+    ENV PHP_FPM_PM_MAX_REQUESTS=500 \
+    docker-php-entrypoint-init php-fpm
```

Comment:
can maybe do `-tt` here then we won't have the `already running` message below?

Reply:
I'll give that a try!

Reply:
Done, check it out!